pkg/tfgen/renames.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2016-2022, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tfgen
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfgen/internal/paths"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/common/tokens"
+)
+
+// Tabulates information about custom inflected or pluralied names. Pulumi bridged providers and Terraform conversion
+// tooling may need to consult these tables to accurately translate between Pulumi and Terraform names.
+type Renames struct {
+	// RenamedResources[t] stores the Terraform name for a resource identified by the Pulumi token t when this name
+	// is not trivially recoverable from the token itself.
```

Comment:
Ah yes, I was trying to find it and it's' both duplicated in a few places and little complicated. Search for this:

```go
// resourceName translates a Terraform name into its Pulumi name equivalent, plus a module name.
func resourceName(provider string, rawname string,
	info *tfbridge.ResourceInfo, isProvider bool) (string, tokens.ModuleName) {
	if isProvider {
		return "Provider", indexMod
	}
	if info == nil || info.Tok == "" {
		// default transformations.
		name := withoutPackageName(provider, rawname)                  // strip off the pkg prefix.
		camel := tfbridge.TerraformToPulumiName(name, nil, nil, false) // camelCase the module name.
		pascal := tfbridge.TerraformToPulumiName(name, nil, nil, true) // PascalCase the resource name.

		modName := tokens.ModuleName(camel)
		return pascal, modName
	}
	// otherwise, a custom transformation exists; use it.
	return info.Tok.Name().String(), info.Tok.Module().Name()
}
```

IN the end I decided to just say ok, whatever, let's just record every resource name instead of mire-optimizing this.